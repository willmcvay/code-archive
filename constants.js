var playerOne = {
	score: 0
}

var playerTwo = {
	score: 0
}

var tiles = [
	a, a, a, a, a, a, a, a, a,
	b, b,
	c, c,
	d, d, d, d,
	e, e, e, e, e, e, e, e, e, e, e, e, 
	f, f,
	g, g, g,
	h, h,
	i, i, i, i, i, i, i, i, i, 
	j,
	k, 
	l, l, l, l,
	m, m,
	n, n, n, n, n, n,
	o, o, o, o, o, o, o, o, 
	p, p,
	q, 
	r, r, r, r, r, r,
	s, s, s, s,
	t, t, t, t, t, t,
	u, u, u, u,
	v, v,
	w, w,
	x, 
	y, y,
	z, 
	blank, blank
]

var tileValues = {
	a: 1,
	b: 3,
	c: 3,
	d: 2,
	e: 1,
	f: 4,
	g: 2,
	h: 4,
	i: 1,
	j: 8,
	k: 5,
	l: 1,
	m: 3,
	n: 1,
	o: 1,
	p: 3,
	q: 10,
	r: 1,
	s: 4,
	t: 1,
	u: 1,
	v: 4,
	w: 4,
	x: 8,
	y: 4,
	z: 10,
	blank: 0
}