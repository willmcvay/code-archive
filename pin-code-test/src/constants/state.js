// @flow
import type { AppState } from '../core/types'

export const DEFAULT_STATE: AppState = {
  currentPin: [],
  displayText: ['*', '*', '*', '*'],
  numberAttempts: 0
}
