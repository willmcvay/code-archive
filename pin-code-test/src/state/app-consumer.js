// @flow
import * as React from 'react'
import { AppContext } from './app-provider'
import type { AppActions, AppState, AppStore } from '../core/types'

type Props = {
  children?: React.Node
}

const AppConsumer: React.StatelessFunctionalComponent<Props> = props => (
  <AppContext.Consumer>
    {context =>
      React.Children.map(props.children, child =>
        React.cloneElement(child, {
          ...context
        })
      )
    }
  </AppContext.Consumer>
)

export default AppConsumer
