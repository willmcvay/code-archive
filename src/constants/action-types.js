// @flow
export const DEFAULT = 'DEFAULT'
export const FETCH_PRODUCTS = 'FETCH_PRODUCTS'
export const FETCH_PRODUCTS_FAILED = 'FETCH_PRODUCTS_FAILED'
export const RECEIVE_PRODUCTS = 'RECEIVE_PRODUCTS'
export const ADD_TO_BASKET = 'ADD_TO_BASKET'
export const REMOVE_FROM_BASKET = 'REMOVE_FROM_BASKET'
export const UPDATE_CARD_NUMBER = 'UPDATE_CARD_NUMBER'
export const SUBMIT_PAYMENT = 'SUBMIT_PAYMENT'
export const CHECKOUT_SUCCESS_PAGE = 'CHECKOUT_SUCCESS_PAGE'
export const CHECKOUT_FAILURE_PAGE = 'CHECKOUT_FAILURE_PAGE'
