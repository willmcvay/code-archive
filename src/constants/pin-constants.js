export const CORRECT_PIN = 1379
