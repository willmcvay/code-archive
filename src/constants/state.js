// @flow
import type { AppState } from '../core/types'

export const DEFAULT_STATE: AppState = {
  currentPin: [],
  lastSelected: null
}
