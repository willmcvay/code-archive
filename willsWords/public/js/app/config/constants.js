define( [ 'App', 'marionette', 'config/constants'],
    function( App, Marionette, constants) {
	var constants = {
		tileValues: {
			a: 1,
			b: 3,
			c: 3,
			d: 2,
			e: 1,
			f: 4,
			g: 2,
			h: 4,
			i: 1,
			j: 8,
			k: 5,
			l: 1,
			m: 3,
			n: 1,
			o: 1,
			p: 3,
			q: 10,
			r: 1,
			s: 4,
			t: 1,
			u: 1,
			v: 4,
			w: 4,
			x: 8,
			y: 4,
			z: 10,
			blank: 0
		},
		rowSquares:[
			[001,002,003,004,005,006,007,008,009,010,011,012,013,014,015],
			[016,017,018,019,020,021,022,023,024,025,026,027,028,029,030],
			[031,032,033,034,035,036,037,038,039,040,041,042,043,044,045],
			[046,047,048,049,050,051,052,053,054,055,056,057,058,059,060],
			[061,062,063,064,065,066,067,068,069,070,071,072,073,074,075],
			[076,077,078,079,080,081,082,083,084,085,086,087,088,089,090],
			[091,092,093,094,095,096,097,098,099,100,101,102,103,104,105],
			[106,107,108,109,110,111,112,113,114,115,116,117,118,119,120],
			[121,122,123,124,125,126,127,128,129,130,131,132,133,134,135],
			[136,137,138,139,140,141,142,143,144,145,146,147,148,149,150],
			[151,152,153,154,155,156,157,158,159,160,161,162,163,164,165],
			[166,167,168,169,170,171,172,173,174,175,176,177,178,179,180],
			[181,182,183,184,185,186,187,188,189,190,191,192,193,194,195],
			[196,197,198,199,200,201,202,203,204,205,206,207,208,209,210],
			[211,212,213,214,215,216,217,218,219,220,221,222,223,224,225]
		],
		columnSquares:[
			[001,016,031,046,061,076,091,106,121,136,151,166,181,196,211],
			[002,017,032,047,062,077,092,107,122,137,152,167,182,197,212],
			[003,018,033,048,063,078,093,108,123,138,153,168,183,198,213],
			[004,019,034,049,064,079,094,109,124,139,154,169,184,199,214],
			[005,020,035,050,065,080,095,110,125,140,155,170,185,200,215],
			[006,021,036,051,066,081,096,111,126,141,156,171,186,201,216],
			[007,022,037,052,067,082,097,112,127,142,157,172,187,202,217],
			[008,023,038,053,068,083,098,113,128,143,158,173,188,203,218],
			[009,024,039,054,069,084,099,114,129,144,159,174,189,204,219],
			[010,025,040,055,070,085,100,115,130,145,160,175,190,205,220],
			[011,026,041,056,071,086,101,116,131,146,161,176,191,206,221],
			[012,027,042,057,072,087,102,117,132,147,162,177,192,207,222],
			[013,028,043,058,073,088,103,118,133,148,163,178,193,208,223],
			[014,029,044,059,074,089,104,119,134,149,164,179,194,209,224],
			[015,030,045,060,075,090,105,120,135,150,165,180,195,210,225]
		]
	}
	return constants;
});

