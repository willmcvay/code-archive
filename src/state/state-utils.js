// @flow
import type { AppState } from '../core/types'
import {
  MAX_ATTEMPTS,
  PIN_LENGTH,
  CORRECT_PIN,
  SUCCESS_MESSAGE,
  LOCKED_MESSAGE,
  ERROR_MESSAGE
} from '../constants/pin-constants'
import { DEFAULT_STATE } from '../constants/state'

const checkPinCorrect = (currentPin: number[]): boolean =>
  currentPin.length === PIN_LENGTH

const isCorrectPin = (currentPin: number[]): boolean =>
  Number(currentPin.join('')) === CORRECT_PIN

const getIsDisabled = (state: AppState): boolean =>
  state.numberAttempts === MAX_ATTEMPTS

const getCurrentPin = (state: AppState, numberTapped: number): number[] =>
  state.currentPin.length < PIN_LENGTH
    ? [...state.currentPin, numberTapped]
    : state.currentPin.length === PIN_LENGTH
      ? [numberTapped]
      : state.currentPin

const getNumberAttempts = (state: AppState, numberTapped: number): number =>
  !checkPinCorrect(getCurrentPin(state, numberTapped))
    ? state.numberAttempts
    : isCorrectPin(getCurrentPin(state, numberTapped))
      ? 0
      : state.numberAttempts < MAX_ATTEMPTS
        ? state.numberAttempts + 1
        : state.numberAttempts

const getPinDisplayString = (currentPin: number[]): string[] =>
  !currentPin.length
    ? DEFAULT_STATE.displayText
    : DEFAULT_STATE.displayText.map(
        (item, index) =>
          index === currentPin.length - 1 ? String(currentPin[index]) : item
      )

const getDisplayText = (state: AppState, numberTapped: number): string[] =>
  isCorrectPin(getCurrentPin(state, numberTapped))
    ? SUCCESS_MESSAGE.split('')
    : getNumberAttempts(state, numberTapped) === MAX_ATTEMPTS
      ? LOCKED_MESSAGE.split('')
      : checkPinCorrect(getCurrentPin(state, numberTapped)) &&
        !isCorrectPin(getCurrentPin(state, numberTapped))
        ? ERROR_MESSAGE.split('')
        : getPinDisplayString(getCurrentPin(state, numberTapped))

export {
  getCurrentPin,
  getDisplayText,
  getIsDisabled,
  getNumberAttempts,
  checkPinCorrect,
  getPinDisplayString
}
