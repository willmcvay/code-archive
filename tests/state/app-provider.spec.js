import React from 'react'
import { shallow } from 'enzyme'
import AppProvider from '../../src/state/app-provider'
import { DEFAULT_STATE } from '../../src/constants/state'
import { LOCKED_TIME } from '../../src/constants/pin-constants'
import * as stateUtils from '../../src/state/state-utils'

describe('AppProvider', () => {
  describe('properties', () => {
    it('should match a DOM snapshot', () => {
      expect(shallow(<AppProvider />)).toMatchSnapshot()
    })

    it('should create a default state', () => {
      expect(shallow(<AppProvider />).state()).toEqual(DEFAULT_STATE)
    })

    it('should return an actions object', () => {
      const componentInstance = shallow(<AppProvider />).instance()

      expect(componentInstance.actions).toEqual({
        numberTapped: componentInstance.numberTapped
      })
    })
  })

  describe('numberTapped', function() {
    const numberTapped = 2

    beforeEach(() => {
      this.getCurrentPinSpy = jest.spyOn(stateUtils, 'getCurrentPin')
      this.getDisplayTextSpy = jest.spyOn(stateUtils, 'getDisplayText')
      this.getNumberAttemptsSpy = jest.spyOn(stateUtils, 'getNumberAttempts')
    })

    it('should call state setters correctly if not disabled', () => {
      this.getIsDisabledSpy = jest
        .spyOn(stateUtils, 'getIsDisabled')
        .mockReturnValue(false)

      const componentInstance = shallow(<AppProvider />).instance()

      componentInstance.numberTapped(numberTapped)

      expect(this.getCurrentPinSpy).toHaveBeenCalledTimes(1)
      expect(this.getCurrentPinSpy).toHaveBeenCalledWith(
        DEFAULT_STATE,
        numberTapped
      )

      expect(this.getDisplayTextSpy).toHaveBeenCalledTimes(1)
      expect(this.getDisplayTextSpy).toHaveBeenCalledWith(
        DEFAULT_STATE,
        numberTapped
      )

      expect(this.getNumberAttemptsSpy).toHaveBeenCalledTimes(1)
      expect(this.getNumberAttemptsSpy).toHaveBeenCalledWith(
        DEFAULT_STATE,
        numberTapped
      )
    })

    it('should not call state setters if disabled', () => {
      this.getIsDisabledSpy = jest
        .spyOn(stateUtils, 'getIsDisabled')
        .mockReturnValue(true)

      const componentInstance = shallow(<AppProvider />).instance()

      componentInstance.numberTapped(numberTapped)

      expect(this.getCurrentPinSpy).not.toHaveBeenCalled()

      expect(this.getDisplayTextSpy).not.toHaveBeenCalled()

      expect(this.getNumberAttemptsSpy).not.toHaveBeenCalled()
    })

    afterEach(() => {
      this.getCurrentPinSpy.mockClear()
      this.getDisplayTextSpy.mockClear()
      this.getNumberAttemptsSpy.mockClear()
      this.getIsDisabledSpy.mockClear()
    })
  })

  describe('locked timer', function() {
    jest.useFakeTimers()

    it('should reset the state to default after 30sec timer if disabled', () => {
      this.getIsDisabledSpy = jest
        .spyOn(stateUtils, 'getIsDisabled')
        .mockReturnValue(true)

      const componentInstance = shallow(<AppProvider />)

      componentInstance.setState({
        currentPin: [1, 2, 3, 4]
      })

      jest.runAllTimers()

      expect(componentInstance.state().currentPin).toEqual(
        DEFAULT_STATE.currentPin
      )
    })

    it('should not reset the state if not disabled', () => {
      this.getIsDisabledSpy = jest
        .spyOn(stateUtils, 'getIsDisabled')
        .mockReturnValue(false)

      const componentInstance = shallow(<AppProvider />)
      const newPin = [1, 2, 3, 4]

      componentInstance.setState({
        currentPin: newPin
      })

      jest.runAllTimers()

      expect(componentInstance.state().currentPin).toEqual(newPin)
    })

    afterEach(() => {
      this.getIsDisabledSpy.mockClear()
    })
  })
})
