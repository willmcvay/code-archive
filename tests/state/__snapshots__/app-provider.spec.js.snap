// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppProvider properties should match a DOM snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AppProvider />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": undefined,
      "value": Object {
        "actions": Object {
          "numberTapped": [Function],
        },
        "state": Object {
          "currentPin": Array [],
          "displayText": Array [
            "*",
            "*",
            "*",
            "*",
          ],
          "numberAttempts": 0,
        },
      },
    },
    "ref": null,
    "rendered": null,
    "type": Object {
      "$$typeof": Symbol(react.provider),
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": [Circular],
        "Provider": [Circular],
        "_calculateChangedBits": null,
        "_changedBits": 0,
        "_currentRenderer": null,
        "_currentValue": Object {
          "currentPin": Array [],
          "displayText": Array [
            "*",
            "*",
            "*",
            "*",
          ],
          "numberAttempts": 0,
        },
        "_defaultValue": Object {
          "currentPin": Array [],
          "displayText": Array [
            "*",
            "*",
            "*",
            "*",
          ],
          "numberAttempts": 0,
        },
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": undefined,
        "value": Object {
          "actions": Object {
            "numberTapped": [Function],
          },
          "state": Object {
            "currentPin": Array [],
            "displayText": Array [
              "*",
              "*",
              "*",
              "*",
            ],
            "numberAttempts": 0,
          },
        },
      },
      "ref": null,
      "rendered": null,
      "type": Object {
        "$$typeof": Symbol(react.provider),
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": [Circular],
          "Provider": [Circular],
          "_calculateChangedBits": null,
          "_changedBits": 0,
          "_currentRenderer": null,
          "_currentValue": Object {
            "currentPin": Array [],
            "displayText": Array [
              "*",
              "*",
              "*",
              "*",
            ],
            "numberAttempts": 0,
          },
          "_defaultValue": Object {
            "currentPin": Array [],
            "displayText": Array [
              "*",
              "*",
              "*",
              "*",
            ],
            "numberAttempts": 0,
          },
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
